### PR TITLE
added websocket server

### PR DESCRIPTION
**Pull request by ben-mkiv**
this required to refactor some of the code to avoid circular dependencies. as a result the CLI code could be greatly reduced and is now pretty simple

* pytaglib was added as requirement, which is now used to keep meta data from the source file and to add meta data with the RVC generation flags to the output file (this could/should probably also be added to batch conversion)
* cli interface was made executable so that it can be run from CLI on linux/mac systems

websocket server listens by default on localhost:15159 (it's NOT recommended to run the websocket server on a public network/interface, since there's no security checks implemented)

example requests:

process an audio file and save the output to
```
{
    "action": "generate",
    "input_audio_path": "/path/to/some/audio.wav",
    "output_path": "/path/to/some/output.wav",
    "model_path": "/path/to/Retrieval-based-Voice-Conversion-WebUI/weights/SomeVoiceWeight.pth"
}
```

get a list of available voice weights
```
{
    "action": "get_voice_weights"
}
```

### CODE REVIEW

File: infer-web.py [hunk 1]
```diff
@@ -70,6 +58,10 @@ def forward_dml(ctx, x, scale):
 mem = []
 if_gpu_ok = False
 
+
+
+start_websocket_server('127.0.0.1', 15159)
```
Comment (fumiama) [inline infer-web.py:63]: You can also add this to config and add a commandline parameter to control this.

File: infer-web.py [hunk 2]
```diff
@@ -18,26 +18,23 @@
 from subprocess import Popen
 from time import sleep
 
+from src.configuration import *
```
Comment (fumiama) [inline infer-web.py:21]: It's better to specify all components used but not `*`.

File: src/utils.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import os
+from src.configuration import weight_root
+
+def get_voice_weights():
```
Comment (fumiama) [inline src/utils.py:4]: maybe this can be put into lib/audio.py

**Review by fumiama — CHANGES_REQUESTED** [review]
The function of the `src` folder is similar to existing `lib` folder, thus all of its contents can be merged into `lib`.

**Comment by ben-mkiv** [comment]
yeah, doesn't make sense to make any changes now if the PR is stall anyways and conflicting with the cleanup which is in progress now...

I'll eventually come back to this when the cleanup is done.

**Comment by fumiama** [comment]
OK. It won't take a long time.

**Pull request closed without merging**